pages/data-providers/[data-provider-id]/deposit-dates.jsx

```diff
@@ -119,7 +119,12 @@ const DepositDates = ({
         <p>
           The chart displays work distribution per deposit time lag in days.
         </p>
-        <TimeLagChart data={store.depositDates.timeLagData} />
+        {Boolean(store.depositDates.timeLagData.length) && (
+          <TimeLagChart data={store.depositDates.timeLagData} />
+        )}
+        {!store.depositDates.timeLagData.length &&
+          !store.depositDates.isRetrieveDepositDatesInProgress &&
```

Comment:
I would simplify it to either `depositDates.loading` or `depositDates.requests > 0` (`depositDates.requestsInProgress > 0`).

To be fair, the data can be in loading, present or absent, so there are 2 levels of conditions:
- if loading, show loading
- else if data show data
- else show message

Reply:
These 2 are mutually excluded but to improve readability I would prefer `?:` at least operator.

However, ideally it would be `isLoading ? 'It\'s loading' : (data.length > 0) ? <Chart /> : 'No data')`

Reply:
Eslint complains about nested ternary operators. That's why I did two separate statements.

Reply:
You can write it all separate but please add loading state reposting if you already started in the beginning.